contributed_definitions/nyaml/NXpositioner_spm.yaml

```diff
@@ -1,242 +1,163 @@
-  velocity(NX_NUMBER):
-    unit: NX_ANY
-    doc: |
-      velocity of the positioner (distance moved per unit time)
-  acceleration_time(NX_NUMBER):
-    unit: NX_ANY
-    doc: |
-      time to ramp the velocity up to full speed
-  controller_record:
-    doc: |
-      Hardware device record, e.g. EPICS process variable, taco/tango ...
-  \@default:
-    doc: |
-      .. index:: plotting
-      
-      Declares which child group contains a path leading
-      to a :ref:`NXdata` group.
-      
-      It is recommended (as of NIAC2014) to use this attribute
-      to help define the path to the default dataset to be plotted.
-      See https://www.nexusformat.org/2014_How_to_find_default_data.html
-      for a summary of the discussion.
-  depends_on(NX_CHAR):
-    doc: |
-      NeXus positions components by applying a set of translations and rotations
-      to apply to the component starting from 0, 0, 0. The order of these operations
-      is critical and forms what NeXus calls a dependency chain. The depends_on
-      field defines the path to the top most operation of the dependency chain or the
-      string "." if located in the origin. Usually these operations are stored in a
-      NXtransformations group. But NeXus allows them to be stored anywhere.
-      
-      .. todo::
-         Add a definition for the reference point of a positioner.
-  (NXtransformations):
-    doc: |
-      This is the group recommended for holding the chain of translation
-      and rotation operations necessary to position the component within
-      the instrument. The dependency chain may however traverse similar groups in
-      other component groups.
-  position(NXtransformations):
-    doc: |
-      To clarify the frame laboratory frame. The scanning area in x, y, and z position in the
-      frame.
-  z_contronller(NX_NUMBER):
+NXpositioner_spm(NXpositioner):
+  z_contronller(NXpid):
     doc: |
       This controllers task is to continuously adjust the Z position of the stm tip in order
       to keep the selected control signal as close as possible to the Set Point. Different control
       signals lead to different contronller beahvior.
+      
+      p_gain (proportional gain) from z_controller is the same as K_p_value from PID controller.
+      i_gain (integral gain) from z_controller is the same as K_p_value from PID controller.
+      setpoint from z_controller is the same as setpoint from PID controller.
+
   z_offset(NX_NUMBER):
     unit: NX_LENGTH
     doc: |
-      Offset added to the initial averaged position Zaver before starting to swepp.
+      Offset added to the initial averaged position tip on Z-axis before starting to sweep.
   tip_position_z(NX_NUMBER):
     unit: NX_LENGTH
     doc: |
       Indicate the tip position Z between tip and sample. The tip position can also be varied when
-      the controller is not running.
+      the controller is not running. This is the actual value (in time) of the tip position.
```

Comment:
If it is values in time, you could thing about using NXlog within NXsensor (see the sensor defined in NXpid).